packages/store-cache/src/utils.ts

```diff
@@ -0,0 +1,194 @@
+
+  // Transform scan args
+  scanArgs.gte = key?.gte && recordToTuple(key.gte);
+  scanArgs.gt = key?.gt && recordToTuple(key.gt);
+  scanArgs.lte = key?.lte && recordToTuple(key.lte);
+  scanArgs.lt = key?.lt && recordToTuple(key.lt);
+
+  // Override gte and lte if eq is set
+  if (key?.eq) {
+    scanArgs.gte = recordToTuple(key.eq);
+    scanArgs.lte = recordToTuple(key.eq);
+  }
+
+  return client.subscribe({ prefix, ...scanArgs }, (write) => {
+    const updates: Record<string, Update> = {};
+
+    // Transform the writes into the expected format
+    for (const update of write.set ?? []) {
+      // The first tuple element is always the table name
+      const table = update.key[0];
+      if (typeof table !== "string") {
+        console.error("Unexpected update:", set);
+        continue;
+      }
+
+      // Add the set event to the updates for this table
+      updates[table] = updates[table] ?? ({ table, set: [], remove: [] } satisfies Update);
+      updates[table].set.push({ key: tupleToRecord(update.key), value: update.value });
+    }
+
+    for (const removedKey of write.remove ?? []) {
+      // The first tuple element is always the table name
+      const table = removedKey[0];
+      if (typeof table !== "string") {
+        console.error("Unexpected update:", removedKey);
+        continue;
+      }
+
+      // Add the remove event to the updates for this table
+      updates[table] = updates[table] ?? ({ table, set: [], remove: [] } satisfies Update);
+      updates[table].remove.push({ key: tupleToRecord(removedKey) });
+    }
+
+    callback(updates as Record<T, Update<C, T>>);
+  });
+}
+
+/**
+ * Map a table schema to the corresponding default value
+ */
+export function getDefaultValue(table?: Record<string, string>) {
+  if (table == null) return undefined;
+
+  // Map schema to its default values
+  const defaultValue: Record<string, unknown> = {};
+  for (const key in table) {
+    defaultValue[key] = AbiDefaults[table[key]];
+  }
+  return defaultValue;
+}
+
+/**
+ * Convert a record like `{ a: string, b: number }` to a record tuple like `[{ a: string }, { b: number }]`,
+ * as expected for keys in tuple-database
+ */
+function recordToTuple(record: Record<string, unknown>): Tuple {
+  const tuple = [];
+  for (const [key, value] of Object.entries(record)) {
+    tuple.push({ [key]: flatten(value) });
+  }
+  return tuple;
+}
+
+/**
+ * Convert a record tuple like `[{ a: string }, { b: number }]` to a record like `{ a: string, b: number }`,
+ */
+function tupleToRecord(tuple: Tuple): Record<string, unknown> {
+  const record: Record<string, unknown> = {};
+
+  for (const entry of tuple) {
+    // Ignore all non-object tuple values
+    if (entry === null || Array.isArray(entry) || typeof entry !== "object") continue;
+    for (const [key, value] of Object.entries(entry)) {
+      record[key] = value;
+    }
+  }
+  return record;
+}
+
+/**
+ * Helper to serialize bigints
+ */
+function flatten(value: unknown) {
```

Comment:
This function is exclusively used in `recordToTuple`, which is exclusively used to turn an object with many keys into an array of objects with one key each, which is the format expected by `tuple-database`. It's only used for "input types", so we don't need to turn the key back into a bigint value (at least for now - if we do need to, we could think about appending `n` to the string and then parsing that in a reverse function, but for now that would be unnecessary complexity).

The type is `unknown` because the function accepts any value and only serializes it if it's a value that's not supported by `tuple-database` (only `bigint` for now). We could just remove this function and do the conversion inline, but that would make the `recordToTuple` function a bit more complex. No strong opinion here though.

For now I changed the name to `serializeKey` and added a better comment to make this a bit more clear!